src/WorkspaceRegistration/WorkspaceRegistration.tsx

```diff
@@ -0,0 +1,299 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Form,
+  Input,
+  Button,
+  Divider,
+  Tooltip,
+  Typography,
+  Space,
+  Result,
+  Radio,
+} from 'antd';
+import { ResultStatusType } from 'antd/lib/result';
+import { Link } from 'react-router-dom';
+
+import './WorkspaceRegistration.css';
+import { userHasMethodForServiceOnResource } from '../authMappingUtils';
+import {
+  hostname, requestorPath, useArboristUI, workspaceRegistrationConfig, kayakoConfig,
+} from '../localconf';
+import { createKayakoTicket } from '../utils';
+import { fetchWithCreds } from '../actions';
+
+const { Text } = Typography;
+
+const layout = {
+  labelCol: {
+    span: 8,
+  },
+  wrapperCol: {
+    span: 32,
+  },
+};
+const tailLayout = {
+  wrapperCol: {
+    offset: 8,
+    span: 16,
+  },
+};
+/* eslint-disable no-template-curly-in-string */
+const validateMessages = {
+  required: '${label} is required',
+};
+/* eslint-enable no-template-curly-in-string */
+
+export interface FormSubmissionState {
+  status?: ResultStatusType;
+  text?: string
+}
+export interface WorkspaceRegistrationProps {
+  userAuthMapping: any
+}
+
+const WorkspaceRegistrationRequestForm: React.FunctionComponent<WorkspaceRegistrationProps> = (props: WorkspaceRegistrationProps) => {
+  const [form] = Form.useForm();
+
+  const [formSubmissionStatus, setFormSubmissionStatus] = useState<FormSubmissionState | null>(null);
+  const [role, setRole] = useState('Principal Investigator');
+  const [formSubmissionButtonDisabled, setFormSubmissionButtonDisabled] = useState(false);
+  const [reqAccessRequestPending, setReqAccessRequestPending] = useState(false);
+
+  useEffect(() => form.resetFields(), [form]);
+
+  const userHasAccess = () => {
+    if (!useArboristUI) {
+      return true;
+    }
+    return (userHasMethodForServiceOnResource('create', 'kayako', '/kayako', props.userAuthMapping));
+  };
+
+  const handleRegisterFormSubmission = (formValues) => {
+    // create a request in requestor
+    const body = {
+      policy_id: workspaceRegistrationConfig?.workspacePolicyId ? workspaceRegistrationConfig.workspacePolicyId : 'workspace',
+    };
+    fetchWithCreds({
+      path: `${requestorPath}request`,
+      method: 'POST',
+      body: JSON.stringify(body),
+    }).then(
+      ({ data, status }) => {
+        if (status === 201) {
+          if (data && data.request_id) {
+            // request created, now create a kayako ticket
+            const fullName = `${formValues['First Name']} ${formValues['Last Name']}`;
+            const email = formValues['E-mail Address'];
+            const subject = 'Registration Access Request for Workspace';
```

Comment:
All request will have the same subject in Kayako?

Reply:
Good point, I reached out to user-services and they suggested that they could use the name of the commons in the title. I've added a new field called [commonsName](https://github.com/uc-cdis/data-portal/pull/1067/commits/eae35aece16e2ae74e6f3744d94f9f5440135612) in localconf.js

Reply:
I honestly think the `hostname` value is representitive enough. 🤷‍♂️  You just need to put it in the subject as well as in the body.

Besides, your logic for generating `commonsName` is not accurate. For example, https://gen3.biodatacatalyst.nhlbi.nih.gov/ will be GEN3, instead of BDCAT

So just use the plain `hostname` instead

Reply:
Thank you. I was only thinking of BRH and QA-BRH when I was writing that logic. This example definitely helps. Will change it back to `hostname`.